code/src/main/webapp/JavaScript/script.js

```diff
@@ -44,17 +44,67 @@ class PageController {
     this.hideCurrentPage();
     switch(pageToShow) {
       case "landing":
-        history.pushState({page: pageToShow}, 'Online Contact Tracing', pageToShow);
-        this.landingPage.show();
-        this.currentlyShown = this.landingPage;
+        var listOfClassesHiddenStatus = {"landing" : false, "login" : true};
+        this.setPageState("landing", this.landingPage, pageToShow, listOfClassesHiddenStatus)
         break;
       case "login":
-        history.pushState({page: pageToShow}, 'Login', pageToShow);
-        this.loginPage.show();
-        this.currentlyShown = this.loginPage;
+        var listOfClassesHiddenStatus = {"landing" : true, "login" : false};
```

Comment:
This will require updating every time we add a new page. Instead, let's do this.

Initialize this varaible above with every page set to true.
in each case statement, set one of the values to "false" (or even better, set the value of pageToShow to false 'listOfClassesHiddenStatus[pageToShow] = false').